### PR TITLE
Extract package subcommand

### PR DESCRIPTION
**Pull request by wolfv**
@jonashaag for reference, this is a simple command implemented as `micromamba package extract myarchive.{tar.bz2,conda}`.

### CODE REVIEW

**Comment by jonashaag** [comment]
Thanks for tagging me!

I'm not sure if `mamba::extract` is the right level of abstraction that a subcommand should be calling into:

- It disables the `extraction_guard` (although I'm not entirely sure what the guard protects)
- It doesn't make too much sense for `.conda` packages IMO, because those are likely to be very quick to extract

In my WIP work (that I didn't push to GitHub so far) I have extracted the body of `extract_archive` to a `extract(archive_file, dest_folder)` routine. The subcommand then calls that one directly, and `extract_archive` calls into the subcommand (after acquiring the guard, if necessary).

**Comment by wolfv** [comment]
`extract(archive_file, dest_folder)` this is coming in #612 as well :)

I think it's still worth to do this for `.conda` files as they might still be big (e.g. if you have `cudatoolkit` and `pytorch` etc.

Locking could be done at the level of the subprocess calling. Or in here. Need to think about it – it's a good point! :)

I am also adding functions to create a archive from a folder and to `transmute` (convert from `.tar.bz2` to `.conda`) a package.

File: src/micromamba/package.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        std::cout << "Extracting " << fs::absolute(infile) << std::endl;
+        extract(fs::absolute(infile));
+    });
+
+    auto compress_subcom = subcom->add_subcommand("compress");
+    compress_subcom->add_option("folder", infile, "Folder to compress");
+    compress_subcom->add_option("dest", dest, "Destination (e.g. myfile-3.1-0.tar.bz2 or .conda)");
+    compress_subcom->add_option("-c,--compression-level",
+                                compression_level,
+                                "Compression level from 0 to 50 (default is 10)");
```
Comment (jonashaag) [inline src/micromamba/package.cpp:32]: Where does that range come from? In the command-line utilities the compression range for bzip2, gzip, zip is 0-9 and the range for zstd is 0-19 IIRC (plus some extra strong compression options that are hidden behind a flag)

Comment (wolfv) [reply src/micromamba/package.cpp:32]: The library supports regular compression levels from 1 up to ZSTD_maxCLevel(),
  which is currently 22. Levels >= 20, labeled `--ultra`, should be used with
  caution, as they require more memory. The library also offers negative
  compression levels, which extend the range of speed vs. ratio preferences.
  The lower the level, the faster the speed (at the cost of compression).

Comment (wolfv) [reply src/micromamba/package.cpp:32]: The above is for zstd. I am actually not sure if compression level is implemented for `tar.bz2` :) let me check

Comment (wolfv) [reply src/micromamba/package.cpp:32]: you're right wrt to bzip2 -- that supports between 0-9 -- I added it as an option.

**Pull request merged**